pkg/internal/cyberark/servicediscovery/discovery_test.go

```diff
@@ -0,0 +1,153 @@
+package servicediscovery
+
+import (
+	"context"
+	"crypto/rand"
+	_ "embed"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/jetstack/preflight/pkg/version"
+)
+
+//go:embed testdata/discovery_success.json
+var discoverySuccessResponse string
+
+func testHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		// This was observed by making a POST request to the integration environment
+		// Normally, we'd expect 405 Method Not Allowed but we match the observed response here
+		w.WriteHeader(http.StatusForbidden)
+		_, _ = w.Write([]byte(`{"message":"Missing Authentication Token"}`))
+		return
+	}
+
+	if !strings.HasPrefix(r.URL.String(), "/services/subdomain/") {
+		// This was observed by making a request to /api/v2/services/asd
+		// Normally, we'd expect 404 Not Found but we match the observed response here
+		w.WriteHeader(http.StatusForbidden)
+		_, _ = w.Write([]byte(`{"message":"Missing Authentication Token"}`))
+		return
+	}
```

Comment:
I get a different response when I use curl to GET that endpoint:

```
$ curl -v https://platform-discovery.integration-cyberark.cloud/api/v2/services/asd | jq
...
> GET /api/v2/services/asd HTTP/2
...
< HTTP/2 400
...
{
  "error": "input validation error"
}
```

Reply:
Ah that's my fault for just making up a random identifier; I'll update this in the Identity client which is to follow!